### ISSUE TITLE
Bug Report: Fetch fails if ansible_ssh_host is localhost

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29572]
From @ghost on 2016-10-18T15:16:29Z
ISSUE TYPE

Bug Report

COMPONENT NAME

fetch

ANSIBLE VERSION

2.1.0.0

CONFIGURATION

These are the most relevant config items though I don't know that there is a correlation:
```
ssh_args = -o ControlMaster=auto -o ControlPersist=60s
```
OS / ENVIRONMENT

N/A

SUMMARY

When ansible_ssh_host is set to localhost, fetch says it succeeds, but it never gets the file. It's important that localhost works the same as every other host value for testing purposes.

I previously opened an issue for this (https://github.com/ansible/ansible-modules-core/issues/4814) but believe it to be closed in error. The symptoms are the same as another issue in which the user didn't have enough space on the target filesystem, but that is not the root cause in this case.

STEPS TO REPRODUCE

Use the play below with ansible_ssh_host set to localhost
```
- hosts: '{{ hosts }}' 
  gather_facts: False 
  tasks:
     - fetch: 
       src: /tmp/remote_file 
       dest: /tmp/local_file 
       flat: true 
       fail_on_missing: true
```
EXPECTED RESULTS

I would expect the behavior to be the same for localhost as it is for every other host.

ACTUAL RESULTS

Fetch says it succeeds, but verbose output actually shows this error (doesn't matter if the file exists or not).
```
ok: [localhost] => {"changed": false, "file": "/tmp/remote_file", "invocation": {"module_args": {"dest": "/tmp/local_file", "fail_on_missing": true, "flat": true, "src": "/tmp/remote_file"}, "module_name": "fetch"}, "msg": "unable to calculate the checksum of the remote file"}
```
Copied from original issue: ansible/ansible-modules-core#5297

**Comment by ansibot** [comment]
From @ghost on 2016-10-18T15:16:29Z

I previously opened an issue for this (#4814) but believe it to be closed in error. The symptoms are the same as another issue in which the user didn't have enough space on the target filesystem, but that is not the root cause in this case.

**Comment by ansibot** [comment]
From @ansibot on 2016-10-18T15:16:29Z

@ansible, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by bcoca** [comment]
I cannot reproduce, i get the file copied as expected and no one else has reported anything similar in a long time:

```
changed: [localhost] => {"changed": true, "checksum": "282003a9cff7bbadb41fa05748c21775eb61dccd", "dest": "/tmp/local_file", "md5sum": "9e7b964750cf0bb08ee960fce356b6d6", "remote_checksum": "282003a9cff7bbadb41fa05748c21775eb61dccd", "remote_md5sum": null}
```

As such I'm going to close this issue, please reopen if you continue see this behavior.
